packages/cli/src/commands/release.js

```diff
@@ -71,15 +73,98 @@ async function release({ bump }) {
   logger.stop();
 
   const branchName = `chore/release-${nextTag}`;
-  logger.start(`Creating branch: ${branchName} from tag ${latestTag}`);
-
-  await execa('git', ['checkout', latestTag]);
+  logger.start(`Creating branch: ${branchName}`);
+
+  if (bump === 'patch') {
+    logger.info(`Using tag ${latestTag} as base`);
+    // If we're bumping for a patch release, we'll need to base our release off
+    // of the previous known tag
+    await execa('git', ['checkout', latestTag]);
+  } else {
+    logger.info(`Using master branch as base`);
+    // If we're publishing other releases, we'll need to base our release off of
+    // the latest stable `master` branch
+    await fetchLatestFromUpstream();
+  }
   await execa('git', ['checkout', '-b', branchName]);
 
   logger.stop();
 
-  const commitRange = `${latestTag}...master`;
+  if (bump === 'patch') {
+    const commitRange = `${latestTag}...master`;
+    await cherryPickCommitsFrom(commitRange, bump);
+  }
+
+  // After making sure our base branch is up-to-date, let's go ahead and reset
+  // our project and rebuild everything from a known state. This is helpful for
+  // getting rid of any local inconsistencies. Ultimately this process
+  // replicates what we do in our Continuous Integration checks.
+  await resetProjectState();
+
+  // Just in case there are any freshly generated files after running the steps
+  // above, we'll check to see if the local branch is dirty before proceeding
+  await checkIfBranchIsDirty();
+
+  // Call out to lerna to handle versioning changed packages
+  await execa(
+    'yarn',
+    ['lerna', 'version', bump, '--no-push', '--no-git-tag-version', '--exact'],
+    {
+      stdio: 'inherit',
+    }
+  );
+
+  logger.start('Creating final commit');
+  logger.info(
+    'The next step will be to manually create a Pull Request for this branch'
+  );
+
+  const versionCommitMessage = 'chore(release): update package versions';
+  await execa('git', ['add', '-A']);
+  await execa('git', ['commit', '-m', versionCommitMessage]);
+
+  logger.stop();
+}
+
+/**
+ * For certain release types, we want to be certain that our base branch is
+ * up-to-date with the upstream remote. This helper will first check that the
+ * upstream remote exists, and create it if it does not, and then will pull the
+ * latest changes into the local project.
+ *
+ * @returns {void}
+ */
+async function fetchLatestFromUpstream() {
+  try {
+    // This command will fail is no upstream is present, with `catch` we can
+    // create the appropriate remote before running the next commands
+    await execa('git', ['remote', 'get-url', 'upstream']);
+  } catch {
+    await execa('git', [
+      'remote',
+      'add',
+      'upstream',
+      'git@github.com:carbon-design-system/carbon.git',
```

Comment:
just curious, is there a reason for using the ssh url over the https url?

Reply:
Ah fair, didn't think about the difference. This is the URL I use by default which is the only reason lol. Would it be helpful to add a prompt to ask which one to use?

Reply:
it's probably fine to just use one or the other, I just noticed that GitHub only exposes the https url on repo pages now